python/langsmith/_internal/_background_thread.py

```diff
@@ -88,6 +92,34 @@ def _tracing_thread_drain_queue(
     return next_batch
 
 
+def _tracing_thread_drain_compressed_buffer(
+    client: Client,
+    runs_limit: int = 100,
+    max_buffer_size: int = 50 * 1024 * 1024
+) -> Optional[bytes]:
+    with client._buffer_lock:
+        current_size = client.tracing_queue.tell()
+
+        # Check if we should send now
+        if not (client._run_count >= runs_limit or current_size >= max_buffer_size):
+            return None
+
+        # Write final boundary and close compression stream
+        client.compressor_writer.write(f'--{client.boundary}--\r\n'.encode())
+        client.compressor_writer.flush()
+        client.compressor_writer.close()
+
+        client.tracing_queue.seek(0)
+        data = client.tracing_queue.getvalue()
```

Comment:
Could be beneficial to stream the data to the api here vs loading the buffer value into memory

Reply:
done!